lnwallet/channel.go

```diff
@@ -6496,6 +6496,19 @@ func (lc *LightningChannel) AvailableBalance() lnwire.MilliSatoshi {
 	return bal
 }
 
+// RemoteHtlcSlots returns the number of slots remaining on the remote party's
+// commitment. This count only includes htlcs that are active on *both*
+// commitment transactions.
+func (lc *LightningChannel) RemoteHtlcSlots() int {
+	lc.RLock()
+	defer lc.RUnlock()
+
+	slots := lc.channelState.LocalChanCfg.MaxAcceptedHtlcs
+	htlcs := len(lc.channelState.ActiveHtlcs())
```

Comment:
`LocalChanCfg.MaxAcceptedHtlcs` is the maximum htlcs we'll accept on our commit tx.
`ActiveHtlcs` only counts htlcs that are active on *both* commit tx's, so it wouldn't count an htlc on the remote's pending commit tx.

Reply:
Related: https://github.com/lightningnetwork/lightning-rfc/issues/745

Reply:
> LocalChanCfg.MaxAcceptedHtlcs is the maximum htlcs we'll accept on our commit tx. 

this got me for a good 5 minutes, "local" is a bit of an ambiguous term here

> ActiveHtlcs only counts htlcs that are active on both commit tx's, so it wouldn't count an htlc on the remote's pending commit tx.

Thanks for pointing this out! Updated to be more accurate and only count outgoing htlcs that we've added to their commitment. I think this approach covers the remote pending commit, but not sure (a bit unfamiliar with this part of the codebase).

Reply:
Just double-checked with a unit test, and `RemoteCommitment` doesn't store pending htlc's.  Those get applied after receiving a revocation from the counterparty.  With [option_dusty_htlcs_uncounted](https://github.com/lightningnetwork/lightning-rfc/issues/873) the way we count htlc's towards max_accepted_htlcs will change, and `RemoteCommitment` also stores dust htlcs.  I think it may be better to avoid duplicating the logic that's already performed in `lnwallet/channel.go` here (especially if it may change in the future):

https://github.com/lightningnetwork/lnd/blob/42099ef5e1e0f927290531fe2bff18406141567b/lnwallet/channel.go#L4938-L4946

I think the lines above can be extracted into a new function that is used to test the max_accepted_htlcs limit, which accounts for the remote pending commitment.

Reply:
Updated to go with the approach suggested here, thanks!

I'm not wild about passing in a "fake" htlc, but don't really see another way around it if we want to use `validateCommitmentSanity`, which makes sense to me.